### PR TITLE
Jamie

### PR DESCRIPTION
**Pull request by wjrossi**
William Rossi, wjr18@pitt.edu

Deliverable 4

### CODE REVIEW

File: com/laboon/ToStringTest.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+		
+		String testString = w.toString();
+		
+		assertEquals(testString.length(), 3);
+	}
+	
+	/*Two worlds constructed with identical conditions should return
+	 * strings of the same length upon invocation of their respective
+	 * toString() methods. World size was limited to 200 maximum
+	 * so as not to destroy my PC with a 999999 by 999999 world.
```
Comment (laboon) [inline com/laboon/ToStringTest.java:30]: good idea

File: com/laboon/ToStringTest.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	 */
+	@Test
+	public void testStringLength() {
+		
+		r = new Random();
```
Comment (laboon) [inline com/laboon/ToStringTest.java:35]: If you use a test double for Random, or set the size, seed, and percent yourself, you can reliably re-create worlds exactly as before.  Would be useful for checking exact output instead of properties.

File: com/laboon/ToStringTest.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	public void testNotNull() {
+		r = new Random();
+		
+		int size = r.nextInt(200);
+		int seed = r.nextInt(10);
+		int pct = r.nextInt(100);
+		
+		w = new World(size, seed, pct);
+		
+		assertNotNull(w.toString());
```
Comment (laboon) [inline com/laboon/ToStringTest.java:101]: Isn't this (non-null returns) checked by other test?

File: com/laboon/ToStringTest.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+	}
+
+
+}
```
Comment (laboon) [inline com/laboon/ToStringTest.java:105]: Could use more checks for exact output and edge cases, but overall good
